xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-macros/xwiki-platform-rendering-macro-rss/src/main/java/org/xwiki/rendering/internal/macro/rss/RssMacro.java

```diff
@@ -112,22 +121,49 @@ public boolean supportsInlineMode()
         return false;
     }
 
+    private void setFeed(SyndFeed feed)
+    {
+        this.execution.getContext().setProperty(FEED_PROPERTY, feed);
+    }
+
+    private SyndFeed getFeed()
+    {
+        return (SyndFeed) this.execution.getContext().getProperty(FEED_PROPERTY);
+    }
+
+    @Override
+    protected ResourceReference getImageReference(RssMacroParameters parameters, String content,
+        MacroTransformationContext context)
+    {
+        if (parameters.isDecoration() && parameters.isImage()) {
+            return new ResourceReference(getFeed().getImage().getUrl(), ResourceType.URL);
+        } else {
+            return super.getImageReference(parameters, content, context);
+        }
+    }
+
+    @Override
+    protected List<? extends Block> getBlockTitle(RssMacroParameters parameters, String content,
+        MacroTransformationContext context)
+    {
+        return generateBoxTitle("rsschanneltitle", getFeed());
+    }
+
+    @Override
+    protected String getClassProperty()
+    {
+        return super.getClassProperty() + " " + FEED_CLASS_VALUE;
+    }
+
     @Override
     public List<Block> execute(RssMacroParameters parameters, String content, MacroTransformationContext context)
         throws MacroExecutionException
     {
         List<Block> result;
         SyndFeed feed = this.romeFeedFactory.createFeed(parameters);
+        this.setFeed(feed);
```

Comment:
Would be nice to explain a bit what we are doing here since it might not be obvious.